src/common/transformations/src/transformations/op_conversions/unique_decomposition.cpp

```diff
@@ -32,9 +44,29 @@ ov::frontend::tensorflow::pass::UniqueReplacer::UniqueReplacer() {
             return false;
         }
 
-        auto x = unique_node->input_value(0);
-        auto output_indices_type = unique_node->get_output_indices_type();
-        auto x_type = x.get_element_type();
+        // currently, the transformation supports only the first and the third outputs
+        if (!unique_node->get_output_target_inputs(1).empty() || !unique_node->get_output_target_inputs(3).empty()) {
+            return false;
+        }
+
+        // currently, the transformation supports only searching for unique elements among scalar elements, not vectors
+        if (unique_node->get_input_size() > 1) {
```

Comment:
The second input is optional and this an axis parameter. When it has the single parameter we can decompose it for sure. Otherwise, we need to check a rank of the input tensor.
Now it is correct.